command_test.go

```diff
@@ -1079,6 +1080,192 @@ func TestHelpExecutedOnNonRunnableChild(t *testing.T) {
 	checkStringContains(t, output, childCmd.Long)
 }
 
+func TestHelpOverrideOnRoot(t *testing.T) {
+	var helpCalled bool
+	rootCmd := &Command{Use: "root"}
+	rootCmd.SetHelpFunc(func(c *Command, args []string) {
+		helpCalled = true
+		if c.Name() != "root" {
+			t.Errorf(`Expected command name: "root", got %q`, c.Name())
+		}
+		if len(args) != 2 || args[0] != "arg1" || args[1] != "arg2" {
+			t.Errorf("Expected args [args1 arg2], got %v", args)
```

Comment:
That's much better.  I started making the change and realized that `slices.Equal` was added in Go 1.18.
Cobra uses go 1.15 to avoid forcing projects to upgrade.
The approach is debatable but is for another discussion 😄 
So I'm forced to leave this change as is.